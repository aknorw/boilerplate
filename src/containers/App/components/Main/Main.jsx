import React from 'react'
import PropTypes from 'prop-types'

import Button from 'components/Button'

import MainWrapper from './MainWrapper'

const Main = ({ currentValue, handleButtonClick }) => (
  <MainWrapper>
    <h1>
      Hello React
    </h1>
    <div>
      <span>
        {`Current value: ${currentValue}`}
      </span>
    </div>
    <Button onClick={handleButtonClick}>
      Change value
    </Button>
  </MainWrapper>
)

Main.propTypes = {
  currentValue: PropTypes.string.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
}

export default Main
