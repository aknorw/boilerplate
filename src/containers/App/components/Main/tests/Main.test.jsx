import React from 'react'
import { shallow } from 'enzyme'

import Main from '../Main'

const mockFn = jest.fn()
console.error = jest.fn()

describe('<Main />', () => {
  it('should be defined', () => {
    expect(Main).toBeDefined()
  })
  it('should log 2 errors when no props are passed', () => {
    shallow(<Main />)
    expect(console.error).toHaveBeenCalledTimes(2)
  })
  it('should render correctly', () => {
    const currentValue = 'current'
    const tree = shallow(
      <Main
        currentValue={currentValue}
        handleButtonClick={mockFn}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
  it('should call mock function when button is clicked', () => {
    const currentValue = 'current'
    const tree = shallow(
      <Main
        currentValue={currentValue}
        handleButtonClick={mockFn}
      />,
    )
    tree.find('Button').simulate('click')
    expect(mockFn).toHaveBeenCalled()
  })
})
