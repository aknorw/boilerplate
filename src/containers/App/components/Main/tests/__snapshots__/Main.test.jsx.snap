// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render correctly 1`] = `
<styled.div>
  <h1>
    Hello React
  </h1>
  <div>
    <span>
      Current value: current
    </span>
  </div>
  <Button
    onClick={[MockFunction]}
  >
    Change value
  </Button>
</styled.div>
`;
